plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoSession.java

```diff
@@ -732,6 +735,14 @@ private static Optional<Object> translateValue(Object trinoNativeValue, Type typ
             return Optional.of(LocalDateTime.ofInstant(instant, UTC));
         }
 
+        if (type == DOUBLE) {
+            return Optional.of(trinoNativeValue);
+        }
+
+        if (type == REAL) {
+            return Optional.of(intBitsToFloat(toIntExact((long) trinoNativeValue)));
+        }
```

Comment:
Move under BIGINT for grouping by numeric types. Also, replace the order as REAL → DOUBLE.